### PR TITLE
Split some helper functions from `load_gltf`

### PR DESCRIPTION
**Pull request by scoopr**
## Checklist

- [ ] cargo clippy reports no issues -- The one clippy issue is unrelated to this change
- [ ] cargo deny issues have been fixed or added to deny.toml
- [ ] relevant examples/test cases run
- [ ] changes added to changelog
  - [ ] Add credit to yourself for each change: `Added new functionality @githubname`.

## Problems PR Solves
The `load_gltf` is adding all the nodes straight to the scene, but often it is desired to just have the meshes be available, and then added in code to the scene as needed.

### CODE REVIEW

**Comment by scoopr** [comment]
Now, I did not check any complicated gltf scenes, so I hope I didn't break any hierarchies there :)

**Comment by scoopr** [comment]
This was perhaps a bit quick'n'dirty, and the interface is not all that fleshed out, and probably needs some kind of a "find mesh_index by name" helper too to be useful. I'm fine iterating on this a bit, if it looks too dirty.

File: rend3-gltf/src/lib.rs [hunk 1]
```diff
@@ -216,49 +204,101 @@ where
...
+}
+
+pub fn load_gltf_nodes<'a, E: std::error::Error + 'static>(
     renderer: &Renderer,
     loaded: &mut LoadedGltfScene,
     nodes: impl Iterator<Item = gltf::Node<'a>>,
     parent_transform: Mat4,
 ) -> Result<Vec<Labeled<Node>>, GltfLoadError<E>> {
-    let mut final_nodes = Vec::new();
+    let mut final_nodes = Vec::with_capacity(nodes.size_hint().0);
```
Comment (cwfitzgerald) [inline rend3-gltf/src/lib.rs:289]: Good spot.

**Review by cwfitzgerald — APPROVED** [review]
Very nice!

**Pull request merged**